evap/staff/importers/enrollment.py

```diff
@@ -0,0 +1,764 @@
+
+    def check_row(self, row: EnrollmentParsedRow) -> None:
+        self.user_data_checker.check_userdata(row.student_data, row.location)
+        self.user_data_checker.check_userdata(row.responsible_data, row.location)
+
+    def finalize(self) -> None:
+        self.user_data_checker.finalize()
+
+
+class CourseDataAdapter(RowCheckerMixin):
+    def __init__(self, course_data_checker) -> None:
+        self.course_data_checker = course_data_checker
+
+    def check_row(self, row: EnrollmentParsedRow) -> None:
+        self.course_data_checker.check_course_data(row.course_data, row.location)
+
+    def finalize(self) -> None:
+        self.course_data_checker.finalize()
+
+
+@transaction.atomic
+def import_enrollments(
+    excel_content, semester: Semester, vote_start_datetime, vote_end_date, test_run: bool
+) -> ImporterLog:
+    # pylint: disable=too-many-locals
+    importer_log = ImporterLog()
+
+    with ConvertExceptionsToMessages(importer_log):
+        input_rows = ExcelFileRowMapper(skip_first_n_rows=1, row_cls=EnrollmentInputRow, importer_log=importer_log).map(
+            excel_content
+        )
+        importer_log.raise_if_has_errors()
+
+        parsed_rows = EnrollmentInputRowMapper(importer_log).map(input_rows)
+        for checker in [
+            TooManyEnrollmentsChecker(test_run, importer_log),
+            CourseDataAdapter(CourseNameChecker(test_run, importer_log, semester=semester)),
+            CourseDataAdapter(CourseDataMismatchChecker(test_run, importer_log)),
+            UserDataAdapter(UserDataEmptyFieldsChecker(test_run, importer_log)),
+            UserDataAdapter(UserDataMismatchChecker(test_run, importer_log)),
+            UserDataAdapter(UserDataValidationChecker(test_run, importer_log)),
+        ]:
+            checker.check_rows(parsed_rows)
+
+        importer_log.raise_if_has_errors()
+
+        user_data_list, course_data_list = normalize_rows(parsed_rows, importer_log)
+        existing_user_profiles, new_user_profiles = get_user_profile_objects(user_data_list)
+
+        responsible_emails = {course_data.responsible_email for course_data in course_data_list}
+        new_responsibles_count = ilen(user for user in new_user_profiles if user.email in responsible_emails)
+        new_participants_count = len(new_user_profiles) - new_responsibles_count
+        new_course_count = ilen(course for course in course_data_list if not course.merge_into_course)
+
+        importer_log.raise_if_has_errors()
+        if test_run:
+            importer_log.add_success(_("The test run showed no errors. No data was imported yet."))
+            msg = format_html(
+                _("The import run will create {evaluation_count} courses/evaluations and {user_count} users:"),
+                evaluation_count=new_course_count,
+                user_count=len(new_user_profiles),
+            )
+            msg += create_user_list_html_string_for_message(new_user_profiles)
```

Comment:
I am slightly in favor of using the result of `create_user_list_html_string_for_message` in the `format_html` itself instead of appending, but you can decide